src/r0/room/report_content.rs

```diff
@@ -0,0 +1,31 @@
+//! [POST /_matrix/client/r0/rooms/{roomId}/report/{eventId}](https://matrix.org/docs/spec/client_server/r0.6.0#post-matrix-client-r0-rooms-roomid-report-eventid)
+
+use js_int::Int;
+use ruma_api::ruma_api;
+use ruma_identifiers::{EventId, RoomId};
+
+ruma_api! {
+    metadata {
+        description: "Report content as inappropriate.",
+        method: POST,
+        name: "report_content",
+        path: "/rooms/:room_id/report/:event_id",
+        rate_limited:  false,
+        requires_authentication: true,
+    }
+
+    request {
+        /// Room in which the event to be reported is located.
+        #[ruma_api(path)]
+        room_id: RoomId,
+        /// Event to report.
+        #[ruma_api(path)]
+        event_id: EventId,
+        /// Integer between -100 and 0 rating offensivness.
+        score: Int,
+        /// Reason to report content. May be blank.
+        reason: Option<String>,
```

Comment:
The docs say that it's required but may be blank. To me that sounds like a `String` that's allowed to be empty, but the field has to be there.

```suggestion
        reason: String,
```

Reply:
Oh, it looks like my previous comment didn't show up. Yeah, I saw that and was thinking that although it is semantically required, in actuality it is optional. I'm working on submitting a proposal to make it optional in the spec. The only positive outcome of making it optional now in `ruma-client-api` is that we won't have to change the API later when if the spec goes through. But since this is bleeding edge anyway, an unstable API isn't an issue.